lib/Controller/BoardApiController.php

```diff
@@ -0,0 +1,146 @@
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Deck\Controller;
+
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IRequest;
+use OCP\IUserManager;
+use OCP\IGroupManager;
+
+use OCA\Deck\Service\BoardService;
+
+/**
+ * Class BoardApiController
+ *
+ * @package OCA\Deck\Controller
+ */
+class BoardApiController extends ApiController {
+
+    private $service;
+    private $userInfo;
+
+    /**
+     * @param string $appName
+     * @param IRequest $request
+     * @param IUserManager $userManager
+     * @param IGroupManager $groupManager
+     * @param BoardService $service
+     * @param $userId
+     */
+    public function __construct($appName, IRequest $request, IUserManager $userManager, IGroupManager $groupManager, BoardService $service, $userId) {
+        parent::__construct($appName, $request);
+        $this->service = $service;
+        $this->userId = $userId;
+        $this->userManager = $userManager;
+        $this->groupManager = $groupManager;
+    }
+
+    /**
+     * @NoAdminRequired
+     * @CORS
+     * @NoCSRFRequired
+     *
+     * Return all of the boards that the current user has access to.
+     */
+    public function index() {
+        $boards = $this->service->findAll($this->getUserInfo());
+
+        return new DataResponse($boards);
+    }
+
+    /**
+     * @NoAdminRequired
+     * @CORS
+     * @NoCSRFRequired
+     *
+     * @params $id
+     *
+     * Return the board specified by $id.
+     */
+    public function get($id) {
+        $board = $this->service->find($id);
+
+        // FIXME: this should probably 404 if the board has been deleted
+
+        return new DataResponse($board);
+    }
+
+    /**
+     * @NoAdminRequired
+     * @CORS
+     * @NoCSRFRequired
+     *
+     * @params $title
+     * @params $color
+     *
+     * Create a board with the specified title and color.
+     */
+    public function create($title, $color) {
+        $board = $this->service->create($title, $this->userId, $color);
+
+        return new DataResponse($board);
+    }
+
+    /**
+     * @NoAdminRequired
+     * @CORS
+     * @NoCSRFRequired
+     *
+     * @params $id
+     *
+     * Delete the board specified by $id.  Return the board that was deleted.
+     */
+    public function delete($id) {
+        $board = $this->service->delete($id);
+
+        return new DataResponse($board);
+    }
+
+    /**
+     * @NoAdminRequired
+     * @CORS
+     * @NoCSRFRequired
+     *
+     * @params $id
+     *
+     * Undo the deletion of the board specified by $id.
+     */
+    public function undoDelete($id) {
+        $board = $this->service->find($id);
+        $this->service->deleteUndo($id);
+
+        return new DataResponse($board);
+    }
+
+    // this is taken from BoardController, but it's not ideal
```

Comment:
You can move the method to the BoardService class, then we can use it from both controllers.

See: https://github.com/nextcloud/deck/blob/2c63bfb4b8a7d5b0c22dc1d753e56d1dd08b61a9/lib/Controller/BoardController.php#L53